smokes_mojo/lib/SmokesMojo/Controller/Api.pm

```diff
@@ -85,6 +88,58 @@ $msg
 EOS
 }
 
+sub post_coresmokedb_report ($self) {
+    my $req = $self->req;
+
+    my $schema = SmokeReports::Dbh->schema;
+    my $sdb = $schema->resultset("Perl5Smoke");
+    my $id = $sdb->get_column("report_id")->max;
+    $id = $id + 1;
+
+    my $json = Cpanel::JSON::XS->new->utf8;
+    my $report_data = $json->decode($req->body);
+    $report_data->{$_} = $report_data->{'sysinfo'}{$_} for keys %{ $report_data->{'sysinfo'} };
+    $report_data->{$_} = $report_data->{'config'}{$_} for keys %{ $report_data->{'config'} };
+    delete $report_data->{'sysinfo'};
+    delete $report_data->{'config'};
+
+    $report_data->{lc($_)} = delete $report_data->{$_} for keys %$report_data;
+
+    $report_data->{smoke_date} = "" . DateTime->from_epoch(
+        epoch     => str2time($report_data->{smoke_date}),
+        time_zone => 'UTC',
+    );
+
+    my @log_data = qw< log_file out_file >;
+    $report_data->{$_} = delete $report_data->{$_} for @log_data;
+
+    my @to_unarray = qw< skipped_tests applied_patches compiler_msgs manifest_msgs nonfatal_msgs >;
+    $report_data->{$_} = join("\n", @{delete($report_data->{$_}) || []}) for @to_unarray;
+
+    my @other_data = qw< harness_only harness3opts summary >;
+    $report_data->{$_} = delete $report_data->{$_} for @other_data;
```

Comment:
This appears to just remove these keys and immediately add them back in with no transformation.